src/main/groovy/com/rundeck/plugins/ansible/ansible/InventoryList.java

```diff
@@ -0,0 +1,166 @@
+package com.rundeck.plugins.ansible.ansible;
+
+import com.dtolabs.rundeck.core.common.NodeEntryImpl;
+import com.dtolabs.rundeck.core.resources.ResourceModelSourceException;
+import lombok.Data;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.lang.String.format;
+
+@Data
+public class InventoryList {
+
+    public static final String ALL = "all";
+    public static final String CHILDREN = "children";
+    public static final String HOSTS = "hosts";
+    public static final String ERROR_MISSING_FIELD = "Error: Missing tag '%s'";
+    public static final String ERROR_MISSING_TAG = "Error: Not tag found among these searched: '%s'";
+
+    /**
+     * Gets value from a tag from Ansible inventory
+     * @param tag tag from yaml output
+     * @param field field to look for
+     * @return value from tag yaml output
+     * @param <T> type you want to receive
+     */
+    public static <T> T getValue(Map<String, Object> tag, final String field) {
+        Object obj = null;
+        if (Optional.ofNullable(tag.get(field)).isPresent()) {
+            obj = tag.get(field);
+        }
+        return getType(obj);
+    }
+
+    /**
+     * Casts an object you want to receive
+     * @param obj generic object
+     * @return cast object
+     * @param <T> type you want to receive
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> T getType(Object obj) {
+        return (T) obj;
+    }
+
+    /**
+     * Process an Ansible tag
+     * @param nodeTag Ansible node tag
+     * @param node Rundeck node to build
+     * @param tags tags to evaluate
+     * @throws ResourceModelSourceException
+     */
+    public static void tagHandle(NodeTag nodeTag, NodeEntryImpl node, Map<String, Object> tags)
+            throws ResourceModelSourceException {
+        nodeTag.handle(node, tags);
+    }
+
+    /**
+     * Finds a tag if it exists in a map of tags
+     * @param nameTags map of tags to find
+     * @param tags tags from Ansible
+     * @return a value if it exists
+     */
+    private static String findTag(List<String> nameTags, Map<String, Object> tags) {
```

Comment:
iterating over and over on the same object doesn't seem optimal, it looks like we have an static set of tags according to the "NodeTag" enum, so i think it would be better to iterate a single time and get all of the needed tags

Reply:
Use a design pattern that avoids the use of nested ifs, which improves performance when there are many variables whose status must be checked. In this case, the different tags that exist from both Ansible and Rundeck.